pymatgen/entries/compatibility.py

```diff
@@ -640,34 +761,228 @@ def __init__(self, compat_type="Advanced", correct_peroxide=True,
-            check_potcar_hash (bool): Use potcar hash to verify potcars are correct.
+            o2_energy: The ground-state DFT energy of oxygen gas, including any adjustments or corrections, in eV/atom.
+                Default: None
+            h2o_energy: The ground-state DFT energy of water, including any adjstments or corrections, in eV/atom.
+                Default: None
+            h2o_adjustments: Total energy adjustments applied to one water molecule, in eV/atom.
+                Default: None
+        """
+        from pymatgen.analysis.pourbaix_diagram import MU_H2O
+        self.MU_H2O = MU_H2O
+        self.o2_energy = o2_energy
+        self.h2o_energy = h2o_energy
+        self.h2o_adjustments = h2o_adjustments
+
+        if not all([self.o2_energy, self.h2o_energy, self.h2o_adjustments]):
+            warnings.warn("You did not provide the required O2 and H2O energies. {} "
+                          "needs these energies in order to compute the appropriate energy adjustments. It will try "
+                          "to infer the values from ComputedEntry for O2 and H2O passed to process_entries, but will "
+                          "fail if these entries are not provided.".format(type(self).__name__))
+
+        # Standard state entropy of molecular-like compounds at 298K (-T delta S)
+        # from Kubaschewski Tables (eV/atom)
+        self.cpd_entropies = {"O2": 0.316731,
+                              "N2": 0.295729,
+                              "F2": 0.313025,
+                              "Cl2": 0.344373,
+                              "Br": 0.235039,
+                              "Hg": 0.234421,
+                              "H2O": 0.071963,  # 0.215891 eV/H2O
+                              }
+        self.name = "MP Aqueous free energy adjustment"
+        super().__init__()
+
+    def get_adjustments(self, entry: ComputedEntry):
         """
-        self.compat_type = compat_type
-        self.correct_peroxide = correct_peroxide
-        self.check_potcar_hash = check_potcar_hash
-        fp = os.path.join(MODULE_DIR, "MPCompatibility.yaml")
-        super().__init__(
-            [PotcarCorrection(MPRelaxSet, check_hash=check_potcar_hash),
-             GasCorrection(fp),
-             AnionCorrection(fp, correct_peroxide=correct_peroxide),
-             UCorrection(fp, MPRelaxSet, compat_type), AqueousCorrection(fp)])
+        Returns the corrections applied to a particular entry.
+
+        Args:
+            entry: A ComputedEntry object.
+
+        Returns:
+            [EnergyAdjustment]: Energy adjustments to be applied to entry.
+
+        Raises:
+            CompatibilityError if the required O2 and H2O energies have not been provided to
+            MaterialsProjectAqueousCompatibility during init or in the list of entries passed to process_entries.
+        """
+        adjustments = []
+        if not all([self.o2_energy, self.h2o_energy, self.h2o_adjustments]):
```

Comment:
Mypy doesn't understand this check. You should do explicit checks for float. Also since the rest of the code is in the else, you can just omit that:
``` python
if not all([isinstance(v,float) for v in [self.o2_energy, self.h2o_energy, self.h2o_adjustments]]):
   raise CompatabilityError(...)

self.h2_energy = ...
```